### ISSUE TITLE
hovering on an overload for  '__new__' not behaving the same as '__init__'

### ISSUE THREAD

**Issue by bschnurr** [microsoft/pyright#4972]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**

Orginal issue found when working with panda's Dataframe.

`overloadsUsedForCall` not being set on `__new__` overloads.

hovering on `Bar([1, 2, 2])` will narrow to a single `__init__` signature but hovering on `Foo([1, 2, 2])` will still show both overloads.

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
class Bar():
    @overload
    def __init__( 
        self,
        data: list[Any]
    ) : ...
    @overload
    def __init__( 
        self,
        data: int,      
    ): ...
    def __init__( 
        self,
        data: int | list[Any],      
    ):
        return super().__init__()


x = Bar([1, 2, 2])  #<- hover on Bar shows a single signature
```
![image](https://user-images.githubusercontent.com/1946977/232924831-9a62a125-c431-485c-ad24-8c06bc30b1fa.png)

```python
class Foo():
    @overload
    def __new__( 
        cls,
        data: list[Any]
    ) : ...
    @overload
    def __new__( 
        cls,
        data: int,      
    ): ...
    def __new__( 
        cls,
        data: int | list[Any],      
    ):
        return super().__new__(cls)


x = Foo([1, 2, 2])    #<- hover on Foo shows multiple signatures
```

![image](https://user-images.githubusercontent.com/1946977/232924792-4034b0d6-96ce-473e-8a2e-ed07e7d88f84.png)

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

**Additional context**

**Comment by erictraut** [comment]
This will be addressed in the next release.
